### PR TITLE
Fix flux calculation in source catalog

### PR DESCRIPTION
**Pull request by larrybradley**
Fixes the flux comparison bug reported #976.  The other issue reported in #976 will require updated NIRCam reference files from the instrument team.

### CODE REVIEW

File: jwst/source_catalog/source_catalog.py [hunk 1]
```diff
@@ -144,24 +144,28 @@ def make_source_catalog(model, kernel_fwhm, kernel_xsize, kernel_ysize,
     catalog['orientation_sky'] = ((270. - rot +
                                    catalog['orientation'].value) * u.deg)
 
-    # define AB mag and AB mag error
+    # define flux in microJanskys
+    nsources = len(catalog)
     pixelarea = model.meta.photometry.pixelarea_arcsecsq
     if pixelarea is None:
-        micro_Jy = 0.0
+        micro_Jy = np.full(nsources, np.nan)
```
Comment (philhodge) [inline jwst/source_catalog/source_catalog.py]: If you don't specify the data type, will this give you the data type that you expect?

Comment (larrybradley) [inline jwst/source_catalog/source_catalog.py]: It returns the data type of the fill value.  In this case it's `np.nan`, which are always float (which is desired/expected).

**Comment by philhodge** [comment]
np.nan gives you float64 rather than float32; that's the distinction I 
was referring to.  I'm all for double precision when it's possible to 
use it, but sometimes you do want single precision.

**Comment by larrybradley** [comment]
Yep, `float64` is what I intended.  Thanks.

**Comment by larrybradley** [comment]
`travis-ci` failures are unrelated...merging.

**Pull request merged**